### PR TITLE
Remove override interface implementations

### LINKED ISSUE #4059: Remove `override` for functions from interfaces & further improvement suggestions (for `v5.0`)
**🧐 Motivation**

All the base implementation contracts use currently the `override` modifiers since Solidity requires the `override` keyword when implementing a function from a parent interface. @frangio once opened an issue [here](https://github.com/ethereum/solidity/issues/8281). A simple example:

```solidity
function totalSupply() public view virtual override returns (uint256) {
    return _totalSupply;
}
``` 

Since Solidity [`0.8.8`](https://github.com/ethereum/solidity/releases/tag/v0.8.8) a function that overrides only a single interface function does not require the `override` modifier anymore. I wanted to quickly discuss what kind of Solidity version is planned for `v5` contracts as well as such a refactoring would be plausible if upgraded to solc `>=0.8.8`.

### PR DESCRIPTION
**Pull request by RenanSouza2**
Fixes #4059

Removes overrider modifier when not needed

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [x] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: .changeset/violet-dancers-cough.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': patch
```
Comment (Amxx) [inline .changeset/violet-dancers-cough.md:2]:
```suggestion
'openzeppelin-solidity': minor
```

**Review by Amxx — DISMISSED** [review]
LGTM, though I'm not sure how to check exhaustivity

File: .changeset/violet-dancers-cough.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'openzeppelin-solidity': minor
+---
+
+Remove override modifier when not needed
```
Comment (ernestognw) [inline .changeset/violet-dancers-cough.md:5]:
```suggestion
Remove the `override` specifier from functions that only override a single interface function.
```

**Review by ernestognw — APPROVED** [review]
I tried to check the missing `override` myself and I didn't find any but I also don't have a way to check exhaustivity. LGTM anyway.

**Review by ernestognw — APPROVED** [review]
I tried to check the missing `override` myself and I didn't find any but I also don't have a way to check exhaustivity. LGTM anyway.

**Comment by ernestognw** [comment]
@RenanSouza2 Can you fix the conflicts? We'll reapprove once done to merge.

**Review by ernestognw — APPROVED** [review]
Thanks @RenanSouza2!

**Review by ernestognw — APPROVED** [review]
Thanks @RenanSouza2!

**Comment by ernestognw** [comment]
Merging since it's in the same state as when approved by @Amxx

**Pull request merged**